### DISCUSSION TITLE
Can't Load Spleeter in Pyodide

### DISCUSSION THREAD

**Discussion by Drzayuss** [pyodide/pyodide#3403] [General]
Hi. I am definitely out of my depth here . . . but I have a working pyodide implementation going in an Angular test environment.  However, I cannot load the spleeter python library (https://pypi.org/project/spleeter/). Here is my code:

--------------------------------
```
ngOnInit() {
      loadPyodide({ indexURL: PYODIDE_BASE_URL }).then((pyodide) => {
      this.pyodide = pyodide;
      this.pyodide.runPython('print("Hello from Python!")');
      this.pyodide.loadPackage("micropip").then(() => {
      this.pyodide.runPython(`import micropip
micropip.install("numpy")`);
      this.pyodide.runPython(`import micropip
micropip.install("spleeter")`);
      this.pyodide.runPython(`import micropip
micropip.install("https://files.pythonhosted.org/packages/55/88/20095d3ccf7d980a8b024df6af5a2d41ce0dd3943aa2f34ccd5960ae52ff/spleeter-2.3.2-py3-none-any.whl")`);
      this.pyodide.loadPackage(
  "https://files.pythonhosted.org/packages/55/88/20095d3ccf7d980a8b024df6af5a2d41ce0dd3943aa2f34ccd5960ae52ff/spleeter-2.3.2-py3-none-any.whl"
);
      

});
      
    
  });
    }


  testButton() {

      this.pyodide.runPython('print("Hello from Python!")');
      this.pyodide.runPython(`package_list = micropip.list()
print(package_list) `);
      this.pyodide.runPython(`import sys
 print("User Current Version:-", sys.version)`);
  }
```
---------------------------

Numpy loads fine . . . but Spleeter does not. One of those attempts to load it should theoretically work. I get an error in the console:

load-pyodide.js:154 Skipping unknown package 'https://files.pythonhosted.org/packages/55/88/20095d3ccf7d980a8b024df6af5a2d41ce0dd3943aa2f34ccd5960ae52ff/spleeter-2.3.2-py3-none-any.whl'

I am obviously missing something . . . or does Spleeter not work with Pyodide?

Thanks in advance for any help you can give me!
Justin

**Comment by ryanking13** [comment]
spleeter contains some non-pure Python dependencies like `llvmbite` and `tensorflow` which is not included in Pyodide. So it won't be possible to install and use spleeter until those dependencies are resolved.
